bootstrap/004_labels.sql

```diff
@@ -55,83 +55,7 @@ BEGIN
 CREATE OR REPLACE VIEW REPORTING.LABELED_QUERY_HISTORY
 COPY GRANTS
 AS
-SELECT  COST,
-        RUN_ID,
-        QTAG,
-        ST,
-        ET,
-        ST_PERIOD,
-        UNLOADED_DIRECT_COMPUTE_CREDITS,
-        DURATION,
-        INCOMPLETE,
-        FILTERTS,
-        QUERY_ID,
-        QUERY_TEXT,
-        DATABASE_ID,
-        DATABASE_NAME,
-        SCHEMA_ID,
-        SCHEMA_NAME,
-        QUERY_TYPE,
-        SESSION_ID,
-        USER_NAME,
-        ROLE_NAME,
-        WAREHOUSE_ID,
-        WAREHOUSE_NAME,
-        WAREHOUSE_SIZE,
-        WAREHOUSE_TYPE,
-        CLUSTER_NUMBER,
-        QUERY_TAG,
-        EXECUTION_STATUS,
-        ERROR_CODE,
-        ERROR_MESSAGE,
-        START_TIME,
-        END_TIME,
-        TOTAL_ELAPSED_TIME,
-        BYTES_SCANNED,
-        PERCENTAGE_SCANNED_FROM_CACHE,
-        BYTES_WRITTEN,
-        BYTES_WRITTEN_TO_RESULT,
-        BYTES_READ_FROM_RESULT,
-        ROWS_PRODUCED,
-        ROWS_INSERTED,
-        ROWS_UPDATED,
-        ROWS_DELETED,
-        ROWS_UNLOADED,
-        BYTES_DELETED,
-        PARTITIONS_SCANNED,
-        PARTITIONS_TOTAL,
-        BYTES_SPILLED_TO_LOCAL_STORAGE,
-        BYTES_SPILLED_TO_REMOTE_STORAGE,
-        BYTES_SENT_OVER_THE_NETWORK,
-        COMPILATION_TIME,
-        EXECUTION_TIME,
-        QUEUED_PROVISIONING_TIME,
-        QUEUED_REPAIR_TIME,
-        QUEUED_OVERLOAD_TIME,
-        TRANSACTION_BLOCKED_TIME,
-        OUTBOUND_DATA_TRANSFER_CLOUD,
-        OUTBOUND_DATA_TRANSFER_REGION,
-        OUTBOUND_DATA_TRANSFER_BYTES,
-        INBOUND_DATA_TRANSFER_CLOUD,
-        INBOUND_DATA_TRANSFER_REGION,
-        INBOUND_DATA_TRANSFER_BYTES,
-        LIST_EXTERNAL_FILES_TIME,
-        CREDITS_USED_CLOUD_SERVICES,
-        RELEASE_VERSION,
-        EXTERNAL_FUNCTION_TOTAL_INVOCATIONS,
-        EXTERNAL_FUNCTION_TOTAL_SENT_ROWS,
-        EXTERNAL_FUNCTION_TOTAL_RECEIVED_ROWS,
-        EXTERNAL_FUNCTION_TOTAL_SENT_BYTES,
-        EXTERNAL_FUNCTION_TOTAL_RECEIVED_BYTES,
-        QUERY_LOAD_PERCENT,
-        IS_CLIENT_GENERATED_STATEMENT,
-        QUERY_ACCELERATION_BYTES_SCANNED,
-        QUERY_ACCELERATION_PARTITIONS_SCANNED,
-        QUERY_ACCELERATION_UPPER_LIMIT_SCALE_FACTOR,
-        TRANSACTION_ID,
-        CHILD_QUERIES_WAIT_TIME,
-        ROLE_TYPE,
-    $$;
+SELECT *,$$;
```

Comment:
I thikn we want to keep the change to remove `select *` unless its absolutely crucial to this migration working. We have had nothing but pain from using select * in past iterations of native app and it doesn't look like it is critical for the migration scheme

Reply:
If we put the column list explicitly in view DDL, I do not see how we can atomically add or drop one column from the view DDL.  One idea is to generate the view DDL by checking the information_schema. However, the generation of view DDL works similar to the auto expansion of `*`, except that it adds more complexities. Sounds to me `select *` or expanding from information_schema is same.

Reply:
If we use `select *` in view DDL,  Snowflake will record the column list expanded from * at the moment view is created.   If the source of `*` is updated, query the  view will hit error. The only way to fix is to re-create the view again.

Here is the example I tried:

- Create a `qh` view with `Col1, Col2`
```
create view qh as select 'col1' as COL1, 'COL2' as COL2;
```

- Create a view `myview`
```
create view myview as select * from qh;
```

- Get the ddl of `myview`. We can see that Col1, Col2 are expanded from * in the view DDL
```
elect get_ddl('view', 'myview');
+--------------------------------+
| GET_DDL('VIEW', 'MYVIEW')      |
|--------------------------------|
| create or replace view MYVIEW( |
| 	COL1,                                                                 |
| 	COL2                                                                 |
| ) as select * from qh;         |
+--------------------------------+
```

-- Change the column list in `qh` from `Col1, Col2`, to `Col3, Col4`.
```
create or replace view qh as select 'col3' as COL3, 'COL2' as COL4;
```

-- Query `myview` hit error
```
select * from myview;
002334 (42601): SQL compilation error: View columns mismatch with view definition for view 'MYVIEW' at line 1, position 14, please re-create the view
```

From this example, `select *` issue can be fixed by re-create the view.  That's what we do in the migration script.  For all the views that depend on `query_history`, we'll re-create those views.

Reply:
right.  We re-create the views in `refresh_queries`, regardless the underlying is changed or not.  The re-creating views will cause the view to have a new expanded column list from `select *` in case underlying is changed. 

I think the column expansion from `select *` in view creating time is same as the column expansion if we choose to construct the view DDL dynamically from information_schema.